src/engine/sparqlExpressions/PrefilterExpressionIndex.cpp

```diff
@@ -162,12 +162,50 @@ static std::string getLogicalOpStr(const LogicalOperator logOp) {
 // SECTION PREFILTER EXPRESSION (BASE CLASS)
 //______________________________________________________________________________
 std::vector<BlockMetadata> PrefilterExpression::evaluate(
-    const std::vector<BlockMetadata>& input, size_t evaluationColumn) const {
+    std::vector<BlockMetadata>& input, size_t evaluationColumn,
+    bool stripIncompleteBlocks) const {
+  if (!stripIncompleteBlocks) {
+    return evaluateAndCheckImpl(input, evaluationColumn);
+  }
+
+  if (input.size() < 3) {
+    return input;
+  }
+  // Helper to check the given block for column consistency up to the evaluation
+  // column.
+  const auto checkBlockIsInconsistent =
+      [evaluationColumn](const BlockMetadata& block) {
+        return getMaskedTriple(block.firstTriple_, evaluationColumn) !=
+               getMaskedTriple(block.lastTriple_, evaluationColumn);
+      };
+  std::optional<BlockMetadata> firstBlock = std::nullopt;
+  std::optional<BlockMetadata> lastBlock = std::nullopt;
+  if (checkBlockIsInconsistent(input.front())) {
+    firstBlock = input.front();
+    input.erase(input.begin());
+  }
+  if (checkBlockIsInconsistent(input.back())) {
+    lastBlock = input.back();
+    input.pop_back();
+  }
```

Comment:
If this is the only reason that you need a mutable reference, then the correct way to do it is to make the
`input` argument a `span<const CompressedBlockMetadata>`. There you can cheaply remove the beginning and the end (via the subspan method).
Otherwise you always have a mutable input argument which you almost never mutate.